libs/elasticsearch/tests/_sync/fake_embeddings.py

```diff
@@ -24,26 +25,33 @@ def embed_query(self, text: str) -> List[float]:
 
 
 class ConsistentFakeEmbeddings(FakeEmbeddings):
-    """Fake embeddings which remember all the texts seen so far to return consistent
-    vectors for the same texts."""
-
-    def __init__(self, dimensionality: int = 10) -> None:
-        self.known_texts: List[str] = []
-        self.dimensionality = dimensionality
+    """Deterministic hash-based embeddings for robust testing. (async version)
```

Comment:
nit: note that due to the way we auto-generate sync code from the async version, it is not possible to say "async version" and "sync version". The comments are identical in both versions.